src/WebApp/Startup.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+
+using Codidact.Authentication.Application.Extensions;
+using Codidact.Authentication.Infrastructure.Extensions;
+
+namespace Codidact.Authentication.WebApp
+{
+    public class Startup
+    {
+        private readonly IConfiguration _configuration;
+
+        public Startup(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddRazorPages()
+                .AddRazorRuntimeCompilation();
+
+            services.AddInfrastructure(_configuration);
+            services.AddApplication(_configuration);
+        }
+
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
```

Comment:
Missing HTTPS redirects, no?

Reply:
Yes, I removed that.

Since we split the project into multiple projects we will need a web-server like Nginx to divide the traffic and send the PII stuff to the authentication server and the other stuff to core. That web-server would also do a number of other things like the URL schema stuff that was discussed a while ago.

In order to do this the web-server would terminate the encryption (at least for the core stuff, for the authentication stuff we can do some magic to avoid this) and then simply send it as HTTP request to the ASP.NET stuff.

Therefor, every request would be a HTTP request which would be redirected to HTTPS creating an infinite loop. I actually have a Nginx configuration setup on my computer thus I already changed this.

I wanted to bring this to attention in a forum post but haven't gotten around doing that yet. If you want I can re-add it, it doesn't really matter for now.

Reply:
Maybe we should do all of this later. I mean use HTTP for development and worry about HTTPS later.

I have changed the authentication server to listen on HTTP only and changed the example client to listen to HTTP too. This would require a small change to core to work (replace `https://` in the authority with `http://`,) but it is so much simpler to run it.

https://discordapp.com/channels/634104110131445811/635510829499940912/681943847579680790

cd70e34715ccb8eedd7e9a13bb12fe28141f2969

I would suggest that we create milestone called something like "Final Sprint" where we add all the issues that would only hinder development but have to be done for production. Then we create an issue "Use HTTPS for Production." and add it to this milestone.

We would start working on that milestone when we get closer to release. I can think of a few more issues that belong into this category, like the URL schema stuff for example.

Reply:
I have no problem with that but I think instead of the `NotImplemented` you can put `UseHsts `and `UseRedirectHttps` because no matter what the solution we will implement we will have to write this here.
Correct me if I am wrong?

Reply:
https://github.com/asynts/codidact-authentication/blob/bb77ebadd11626a40ee14357be1e0b6230a50ae1/src/WebApp/Startup.cs#L45-L52

You want me to replace that with the following?

~~~csharp
if (env.IsProduction())
{
    app.UseHsts();
    app.UseRedirectHttps();
}
else
{
     app.UseDeveloperExceptionPage();
}
~~~

Not sure if this is a good idea. There are a lot of things that have to be implemented before we can switch to production. Like `ISecretsService`, HTTPS, a web-server, *et cetera*. The exceptions sends the clear message, that you are not supposed to enable production yet.

Reply:
I have done it a bit different, now the 'Development' and 'Production' environments are allowed, choosing any other environment will throw an exception because that environment was not configured yet.

08252e49534558719edd51e3af56937ad2ad13cf